benches/dapol.rs

```diff
@@ -1,175 +1,250 @@
-use core::fmt::Debug;
-use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion, SamplingMode};
-use dapol::{
-    utils::get_secret, Dapol, DapolNode, RangeProofPadding, RangeProofSplitting, RangeProvable,
-    RangeVerifiable,
-};
-use digest::Digest;
-use rand::{distributions::Uniform, thread_rng, Rng};
-use smtree::{
-    index::TreeIndex,
-    traits::{ProofExtractable, Rand, Serializable, TypeName},
-};
-use std::time::Duration;
-
-// CONSTANTS
-// ================================================================================================
+use std::path::PathBuf;
 
-const TREE_HEIGHTS: [usize; 3] = [16, 24, 32];
-const NUM_LEAVES: [usize; 3] = [1024, 2048, 4096];
+use criterion::{criterion_group, criterion_main};
+use criterion::{BenchmarkId, Criterion, SamplingMode};
+use jemalloc_ctl::{epoch, stats};
 
-// BENCHMARKS
-// ================================================================================================
+use dapol::accumulators::NdmSmt;
+use dapol::{EntityId, Height, InclusionProof, MaxThreadCount};
+
+mod setup;
+use crate::setup::{MemoryUsage, Metrics, Variable, NUM_USERS, TREE_HEIGHTS};
 
-fn build_dapol(c: &mut Criterion) {
-    let mut group = c.benchmark_group("build");
+#[global_allocator]
+static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
+
+fn bench_dapol(c: &mut Criterion) {
+    let e = epoch::mib().unwrap();
+    let alloc = stats::allocated::mib().unwrap();
+    let act = stats::active::mib().unwrap();
+    let res = stats::resident::mib().unwrap();
+
+    let mut group = c.benchmark_group("dapol");
     group.sample_size(10);
     group.sampling_mode(SamplingMode::Flat);
```

Comment:
Add a comment here saying that `Flat` is used because that is what is recommended by Criterion for long-running benches: https://bheisler.github.io/criterion.rs/book/user_guide/advanced_configuration.html#sampling-mode